### ISSUE TITLE
cstat goodness of fit approximation from Kaastra

### ISSUE THREAD

**Issue by vkashyap** [sherpa/sherpa#594]
I have a prototype function that can compute the expected null cstat for the best-fit model which should be easily translatable to python and used in Sherpa.  See
https://github.com/vkashyap/PINTofALE/blob/master/pro/stat/cstat_gof_k17.pro

**Comment by anetasie** [comment]
Thanks! I guess we need a volunteer to translate this into python.

**Comment by DougBurke** [comment]
And also some ide of how / when to use it (for documentation).

**Comment by vkashyap** [comment]
It is useful whenever a cstat is calculated.  Just echo a "for this model, to a ballpark approximation, a good fit is expected to result in a cstat value of mean±error"

**Comment by anetasie** [comment]
@vkashyap sent a python version of the cstat gof calculations with a simple working example. 

> The cstat for a good fit to the model should be distributed as
> N(expected cstat, expected variance).   Large values of |(observed
> cstat - expected cstat)/sqrt(expected variance)| means fits are bad.
> Small values don't say anything one way or the other.  Very similar to
> chisq/nu goodness of fit estimates.

The script and the working example will need to be converted to a Sherpa function that returns `observed cstat, expected cstat, expected cstat variance` and the `cstat_gof  |(observed cstat - expected cstat)/sqrt(expected variance)| `

Here is the example of how to use it, which involves getting the data and model into arrays to calculate the gof using cstat_gof.py script. Note I attach the script with TXT extension, so github takes it.

[cstat_gof.py.txt](https://github.com/sherpa/sherpa/files/5070217/cstat_gof.py.txt)

```
import cstat_gof
from sherpa.astro.ui import  * 

plot_data()
xmid=get_data_plot().x
yval=get_data_plot().y
xlo=get_model_plot().xlo
xhi=get_model_plot().xhi
exptim=get_data().exposure
plot_model()
ymod=get_model_plot().y
miu=ymod*exptim*(xhi-xlo)
obs=yval*exptim*(xhi-xlo) #verify that obs are now all integer values
cstat_gof.gof_cstat(miu,obs)

```

**Comment by anetasie** [comment]
@DougBurke can this be made available as an additional function/script?

**Comment by anetasie** [comment]
goodness-of-fit calculations in the prototype are based on the Kaastra 2017 paper: https://ui.adsabs.harvard.edu/abs/2017A%26A...605A..51K/abstract

**Comment by anetasie** [comment]
@DougBurke the prototype can be used as a script. The functionality is mainly relevant to X-ray analysis, so we could included it as part of CIAO contrib scripts package.

**Comment by DougBurke** [comment]
Too lazy to look at the paper - does it include support for modelling the background (what we call "wstat")?

**Comment by vkashyap** [comment]
No.  It relies entirely on the defined model, which may include the
background but not necessarily
